website/docs/docs/build/unit-tests.md

```diff
@@ -0,0 +1,139 @@
+
+Let’s say you’re creating a new `dim_customers` model with a field `is_valid_email_address`, that calculates whether or not the customer’s email is valid: 
+
+```sql
+with customers as (
+
+    select * from {{ ref('stg_customers') }}
+
+),
+
+accepted_email_domains as (
+
+    select * from {{ ref('top_level_email_domains') }}
+
+),
+	
+check_valid_emails as (
+
+    select
+        customers.customer_id,
+        customers.first_name,
+        customers.last_name,
+				coalesce (regexp_like(
+            customers.email, '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$'
+        )
+        = true
+        and accepted_email_domains.tld is not null,
+        false) as is_valid_email_address
+    from customers
+		left join accepted_email_domains
+        on customers.email_top_level_domain = lower(accepted_email_domains.tld)
+
+)
+
+select * from check_valid_emails
+```
+
+This type of logic can be challenging to validate. Let’s add a unit test to this model to ensure our `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
+
+```yaml
+unit_tests:
+  - name: test_is_valid_email_address
+    description: "Check my is_valid_email_address logic captures all known edge cases - emails without ., emails without @, and emails from invalid domains."
+    model: dim_customers
+    given:
+      - input: ref('stg_customers')
+        rows:
+          - {customer_id: 1, email: cool@example.com, email_top_level_domain: example.com}
+          - {customer_id: 2, email: cool@unknown.com, email_top_level_domain: unknown.com}
+          - {customer_id: 3, email: badgmail.com, email_top_level_domain: gmail.com}
+          - {customer_id: 4, email: missingdot@gmailcom, email_top_level_domain: gmail.com}
+      - input: ref('top_level_email_domains')
+        rows:
+          - {tld: example.com}
+          - {tld: gmail.com}
+    expect:
+      rows:
+        - {customer_id: 1, is_valid_email_address: true}
+        - {customer_id: 2, is_valid_email_address: false}
+        - {customer_id: 3, is_valid_email_address: false}
+        - {customer_id: 4, is_valid_email_address: false}
+
+```
+
+The above example defines the mock data using the inline `dict` format, but there are a handful of different options for how you format your mock data. 
+
+You’ll notice that we _only_ had to define the mock data for the columns we care about. This enables you to write succinct and _specific_ unit tests.
+
+:::note
+
+The direct parents of the model that you’re unit testing (in this example `stg_customers` and `top_level_email_domains`) need to exist in the warehouse before you’re able to execute the unit test.
+
+Use the `—-empty` flag to build an empty version of those models to save warehouse spend. 
```

Comment:
Once we have docs for this flag, we should link to them!

Reply:
<img width="228" alt="Screenshot 2024-02-09 at 8 54 18 AM" src="https://github.com/dbt-labs/docs.getdbt.com/assets/53586774/a4fe3ec3-614b-4872-a228-a0dba19513a8">

the double dashes look a bit odd her for some reason